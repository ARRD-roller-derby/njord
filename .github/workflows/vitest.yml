name: Vitest
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v2
      with:
        node-version: '16.x'
    - name: Install dependencies
      run: npm ci
    - name: Run Vitest tests
      run: npm run coverage
    - uses: actions/upload-artifact@v2
      if: always()
      with:
        name: vitest-report
        path: vitest-report/
        retention-days: 30