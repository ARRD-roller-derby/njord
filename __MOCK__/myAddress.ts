import dayjs from 'dayjs'
import { addressInterface } from '../types/address.interface'
import { addressType } from '../types/addressType.enum'
export const myAddress: addressInterface = {
  _id: 'fake_id',
  label: 'my address',
  lat: 2.222222,
  lon: 3.333333,
  ownerId: 'my_id',
  city: 'Gotham City',
  zipcode: '010101',
  street: 'Rue du crime',
  type: addressType.personnal,
  updatedAt: dayjs('01/01/2022').toDate(),
}
