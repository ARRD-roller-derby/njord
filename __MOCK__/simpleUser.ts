import { UserInterface } from "../types/User.interface";

export const simpleUser: UserInterface = {
  _id: 'simple_user_id',
  email: 'simple@user.fr',
  name: 'simple',
  lastname: 'user',
  derbyName: 'mega shark',
  numRoster: '01',
  msp: false,
  mst: true,
  profiles: [],
  dailyContestAvgTime: 13,
  dailyContestAvgAccuracy: 50,
  league: {
    id: 'my_league_id',
    shortName: 'my league'
  }
}