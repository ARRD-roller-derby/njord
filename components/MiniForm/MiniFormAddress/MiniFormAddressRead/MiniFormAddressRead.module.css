.empty {
  opacity: 0.5;
}

.container {
  display: flex;
  flex-direction: column;
  gap: 5px;
}

.label {
  color: var(--color-tierce);
}

.street {
  font-weight: bold;
  text-transform: uppercase;
}

.city {
  font-style: italic;
}