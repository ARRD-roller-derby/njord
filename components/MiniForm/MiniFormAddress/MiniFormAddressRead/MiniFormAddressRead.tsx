import classes from './MiniFormAddressRead.module.css'

interface props {
  readonly value?: string
}

export default function MiniFormAddressRead({ value }: props) {
  return <>{value || <span className={classes.empty}>{'(vide)'}</span>}</>
}
