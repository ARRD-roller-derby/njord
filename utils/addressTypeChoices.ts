import { addressType } from "../types/addressType.enum";

export const addressTypeChoices = [
  {
    label: 'Mon adresse',
    value:addressType.personnal
  },
  {
    label: 'Stade',
    value:addressType.stadium
  },
  {
    label: 'Point de covoiturage',
    value:addressType.carpool
  },
  {
    label: 'Autre',
    value:addressType.other
  }
]