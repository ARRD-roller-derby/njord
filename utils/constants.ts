export const DISCORD_NEWS_HOOK = process.env.DISCORD_NEWS_HOOK || ''