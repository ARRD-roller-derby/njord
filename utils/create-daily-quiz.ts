import dayjs from "dayjs"
import Question from "../models/question.model"
import Quiz from "../models/quiz.model"
import { QuizType } from "../types/quiz.interface"
import { percent } from "./percent"
import { questionDifficulty } from "./question-difficulty"
import { shuffle } from "./shuffle"
import { MongoDb } from '../db/mongo.connect'

export async function createDailyQuiz() {

  await MongoDb()

  const existDailyQuiz = await Quiz.findOne({
    type: QuizType.daily,
    day: dayjs().format('YYYY-MM-DD')
  })

  if (existDailyQuiz) return existDailyQuiz

  // Search old Quizzes
  const oldsQuizzes = await Quiz.find({
    type: QuizType.daily, day: {
      $in: [
        dayjs().subtract(4, 'day').format('YYYY-MM-DD'),
        dayjs().subtract(3, 'day').format('YYYY-MM-DD'),
        dayjs().subtract(2, 'day').format('YYYY-MM-DD'),
        dayjs().subtract(1, 'day').format('YYYY-MM-DD')
      ]
    },
    answser: { $exists: true, $not: { $size: 0 } },
  }).select('questions')

  //Take olds questions
  const oldsQuestions = oldsQuizzes.reduce((current: string[], state) => {
    current.push(...state.questions)
    return current
  }, [])

  //Take new questions without olds questions
  const questions = await Question.find({ active: true, _id: { $not: { $in: oldsQuestions } } }).select('_id bad_answers_num good_answers_num')
  const shuffleQuestions = shuffle<string>(questions.map(question => question._id), 4)

  const difficulties = questions
    .filter((question) => shuffleQuestions.includes(question._id))

  //For difficulty
  const questionsNum = difficulties.reduce((current: { bad: number, good: number }, state) => {
    current.bad += state.bad_answers_num
    current.good += state.good_answers_num
    return current
  }, { bad: 0, good: 0 })

  return await Quiz.create({
    difficulty: questionDifficulty(
      percent(
        questionsNum.good,
        questionsNum.good + questionsNum.bad
      )
    ),
    questions: shuffleQuestions,
    type: QuizType.daily,
    day: dayjs().format('YYYY-MM-DD')
  })

} 