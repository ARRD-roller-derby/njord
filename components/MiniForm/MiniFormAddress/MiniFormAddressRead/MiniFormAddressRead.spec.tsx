import { afterEach,describe,expect, it, vi } from 'vitest'
import { cleanup } from '@testing-library/react'
import MiniFormAddressRead from './MiniFormAddressRead'
import { render, screen } from '../../../../utils/test-utils'
import { myAddress } from '../../../../__MOCK__/myAddress'

describe('<MiniFormAddressRead />', () => {
  afterEach(cleanup)

  it('Check snapshot', async () => {
    const {asFragment } = render(<MiniFormAddressRead/>
    )
    expect(await screen.findByText("(vide)")).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  it('Check snapshot with value', async () => {
    const {asFragment } = render(<MiniFormAddressRead value={myAddress}/>
    )
    expect(await screen.findByText("010101 - Gotham City")).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
