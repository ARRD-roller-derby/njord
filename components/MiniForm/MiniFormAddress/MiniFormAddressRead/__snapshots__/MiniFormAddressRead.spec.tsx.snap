// Vitest Snapshot v1

exports[`<MiniFormAddressRead /> > Check snapshot 1`] = `
<DocumentFragment>
  <span
    class="_empty_xerbv_1"
  >
    (vide)
  </span>
</DocumentFragment>
`;

exports[`<MiniFormAddressRead /> > Check snapshot with value 1`] = `
<DocumentFragment>
  <div
    class="_container_xerbv_5"
  >
    <div
      class="my address"
    >
      my address
    </div>
    <div
      class="Rue du crime"
    >
      Rue du crime
    </div>
    <div
      class="Gotham City"
    >
      010101 - Rue du crime
    </div>
  </div>
</DocumentFragment>
`;
